corehq/apps/reports/standard/cases/basic.py

```diff
@@ -68,14 +70,31 @@ def _build_query(self):
             )
             query = query.NOT(case_es.owner(ids_to_exclude))
         else:  # Only show explicit matches
-            query = query.owner(self.case_owners)
+            selected_case_owners = self.case_owners
+            if selected_case_owners:
+                query = query.owner(selected_case_owners)
```

Comment:
When you deselect everything, the current behavior is to not return any results.  This changes it to return all results, unfiltered (except by scope, where applicable).  I don't think it matters a lot which one we do @sheelio do you have any thoughts?

One possible point of confusion is that if you select something like "Unknown Users", and there are no unknown users on the project, `case_owners` will return `[]`, and then the report will confusingly render with all cases.  I could see users thinking that means that their data is messed up.